### PR TITLE
feat: enable role-based permissions for Netlify CMS (fix #470)

### LINKED ISSUE #470: Restrict CMS access to specific Netlify Identity role
## Is your feature request related to a problem? Please describe.

Our Netlify account gives us 1000 Netlify Identity users but only 5 invitation-only users. However, we need >5 CMS users for the We Count site.

## Describe the solution you'd like

We need to assign an `admin` role to Netlify Identity users with the appropriate permissions and block/redirect users without that role from accessing the CMS using the technique described [here](https://docs.netlify.com/visitor-access/role-based-access-control/#redirect-visitors-based-on-roles).

## Describe alternatives you've considered

Not applicable.

## Additional context

Not applicable.

### PR DESCRIPTION
**Pull request by greatislander**
* [x] This pull request has been linted by running `npm run lint` without errors
* [x] This pull request has been tested by running `npm run start` and reviewing affected routes
* [x] This pull request has been built by running `npm run build` without errors
* [x] This isn't a duplicate of an existing pull request

## Description

Returns 404 for non-admin users when visiting `/admin/`.

## Steps to test

1. Ensure that your user has the `admin` role in the Netlify Identity dashboard.
1. Log in via the log in link on the home page (at the bottom).
2. Visit `/admin/`.
3. Log out.
4. From the Netlify Identity dashboard, remove the `admin` role from your user.
5. Log in again.
6. Visit `/admin/`.

**Expected behavior:**

1. With the `admin` role, `/admin` returns the Content Management System.
2. Without the `admin` role, `/admin` returns a 404 error.

## Additional information

Not applicable.

## Related issues

Not applicable.

### CODE REVIEW

**Comment by cindyli** [comment]
Accessing https://deploy-preview-471--wecount.netlify.app/admin gives "Page not found" error. I was expecting to be redirected to a login page. Do I miss anything?

**Comment by greatislander** [comment]
> Accessing https://deploy-preview-471--wecount.netlify.app/admin gives "Page not found" error. I was expecting to be redirected to a login page. Do I miss anything?

You have to follow this instruction as in my comment:

> Log in via the log in link on the home page (at the bottom).

The redirect checks for the role. If you aren't logged in, there's no role so the redirect will be in effect. The process with this new check is:

1. Log in from the home page.
2. Navigate to the admin page.

**Comment by cindyli** [comment]
Thanks for the clarification, @greatislander. I login from the home page. When navigating to the admin page, the error "Failed to load config.yml (404)" appears. My role on the Netlify identity is "admin" at the moment.

error

**Comment by greatislander** [comment]
@cindyli Can you try refreshing/clearing your cache? I saw that initially as well.

**Comment by cindyli** [comment]
Editing the cover image has the same issue as running Netlify CMS on the local machine: if the image source is an url, the preview pane will be able to display it. If the image is uploaded from the local drive, the preview cannot show it.

If this is a known issue, we should document it in the documentation for content editors. Ned, do you think if your in-progress [Netlify CMS tips & tricks page](https://wiki.fluidproject.org/pages/viewpage.action?pageId=228360215) a right place for it?

**Comment by greatislander** [comment]
@cindyli Yes, I can document it there. Can you create a separate issue in this repo and assign it to me?

**Comment by cindyli** [comment]
> Can you create a separate issue in this repo and assign it to me?

The issue is created at https://github.com/inclusive-design/wecount.inclusivedesign.ca/issues/472.

**Comment by cindyli** [comment]
After login, when accessing the admin page, if the user appends the URL with "/admin", the error `Failed to load config.yml (404)` will appear. However, if the URL is appended with "/admin/" with the ending slash, the admin page will show up without any problem. The ending slash is the magic character. :)

**Comment by greatislander** [comment]
> After login, when accessing the admin page, if the user appends the URL with "/admin", the error `Failed to load config.yml (404)` will appear. However, if the URL is appended with "/admin/" with the ending slash, the admin page will show up without any problem. The ending slash is the magic character. :)

From Netlify's [redirect docs](https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash):

> You cannot use a redirect rule to add or remove a trailing slash. However, you can enable Netlify’s **Pretty URLs** feature in **Site settings > Build & deploy > Post processing > Asset optimization**.

I tried that and it didn't work. I also tried adding a separate redirect rule for `/admin` without the trailing slash and that didn't work either. I think content editors just need to use the full URL with the slash.

**Comment by cindyli** [comment]
> From Netlify's [redirect docs](https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash):
> 
> > You cannot use a redirect rule to add or remove a trailing slash. However, you can enable Netlify’s **Pretty URLs** feature in **Site settings > Build & deploy > Post processing > Asset optimization**.
> 
> I tried that and it didn't work. I also tried adding a separate redirect rule for `/admin` without the trailing slash and that didn't work either. I think content editors just need to use the full URL with the slash.

Thanks for trying various approaches, @greatislander. Ideally, when someone logs in as an admin, If an admin link/button can show somewhere and directs the person to the admin page, it would be more convenient. Asking content editors to manually update the URL to access the admin page is not ideal but would be OK as we are using the same approach with WordPress. 

BTW, the emphasis of the ending slash is a good addition to the documentation too.

**Comment by greatislander** [comment]
Can you open a separate issue to display a link to the admin page for all admin users?

File: eleventy.config.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ module.exports = function(eleventyConfig) {
 		sourcemaps: process.env.ELEVENTY_ENV === "development" ? true : false
 	});
 	eleventyConfig.addPlugin(pluginPWA, {
-		globIgnores: ["admin/*"]
+		globIgnores: ["admin/*", "login/*"]
```
Comment (cindyli) [inline eleventy.config.js:119]: We don't have a login page or login URL. Is "login/*" needed?

**Comment by cindyli** [comment]
> Can you open a separate issue to display a link to the admin page for all admin users?

The issue is created at https://github.com/inclusive-design/wecount.inclusivedesign.ca/issues/475.

**Review by cindyli — APPROVED** [review]

**Pull request merged**